### ISSUE TITLE
Dynaconf returns a fresh new object every time a list/dict is accessed.  was: boxlist.append() is not working

### ISSUE THREAD

**Issue by sla-te** [dynaconf/dynaconf#568]
If I have a toml config file with ITEM_INNER_SPACING = [ 4.00, 4.00 ] and then do e.g. settings.ITEM_INNER_SPACING.append(1), it does not seem to append 1 to the existing values.

**Comment by rochacbruno** [comment]
You found a bug, BoxList.append is not working.

https://replit.com/@rochacbruno/EvergreenDimpledSolaris

```py

from dynaconf import Dynaconf

settings = Dynaconf(settings_file="settings.toml")

print(settings.ITEM_INNER_SPACING)
assert settings.ITEM_INNER_SPACING == [4.0, 4.0]


# this is not working
settings.ITEM_INNER_SPACING.append(1)  

# This is the workaround
settings.set("ITEM_INNER_SPACING", [1, "dynaconf_merge"])


print(settings.ITEM_INNER_SPACING)
assert settings.ITEM_INNER_SPACING == [4.0, 4.0, 1]
```

**Comment by lbmendes** [comment]
I've tried to work on this and i noticed that **inside** the append function the object is updated with the new value.

But everytime i call settings.ITEM_INNER_SPACING the boxlist is recreated from scratch with a loop reinserting one value at a time and the value appended before (1) is not being reinserted. I'm a novice dev and don't know where to go from here.

**Comment by andressadotpy** [comment]
Hi @lbmendes! I'm investigating this and I have some ideas about it. Do you wanna pair with me in this issue? We can practice some debugging and I can talk about what I've been learning here in Dynaconf project. If you're interested, let me know!

**Comment by pvmm** [comment]
The **evaluate_lazy_format()** function creates a new BoxList every time you access the variable, so you are not using the same object after you appended a value at the end.

**Comment by rochacbruno** [comment]
This is issue is related to #690 and all other "Dynamic Evaluation" related issues.

The way lazy evaluation is implemented does a recursive evaluation every time the inner value is accessed and tries to resolve templates, so it always returns a new object.

When we do: `settings.a_list.append(x)` the `append` is working properly, however the object returned by `settings.a_list` is not always the same when the key holds lists and dics dynaconf is traversing and creating a whole new object.

This algorithm needs change

**Comment by rochacbruno** [comment]
this issue is going to be fixed on #1128 and #1129 as we are going to replace Box objects
